test/e2e/snat/snat_test.go

```diff
@@ -0,0 +1,160 @@
+package snat
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	TEST_POD_LABEL_KEY   = "test-pod-label-key"
+	TEST_POD_LABEL_VALUE = "test-pod-label-val"
+	EXTERNAL_DOMAIN      = "https://aws.amazon.com/"
+)
+
+var _ = Describe("SNAT test", func() {
+	It("Pod in private subnet should have Internet access with External SNAT disabled", func() {
+		By("Setting External SNAT to false")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT": "false",
+		})
+
+		By("Checking External Domain Connectivity")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
+	})
+
+	It("Pod in private subnet should have Internet access with External SNAT enabled", func() {
+		By("Setting External SNAT to true")
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, utils.AwsNodeName, utils.AwsNodeNamespace, utils.AwsNodeName, map[string]string{
+			"AWS_VPC_K8S_CNI_EXTERNALSNAT": "true",
+		})
+
+		By("Checking External Domain Connectivity")
+		ValidateExternalDomainConnectivity(EXTERNAL_DOMAIN)
```

Comment:
I think we can make this more concrete by verifying by setting `AWS_VPC_K8S_CNI_EXTERNALSNAT` SNAT rules are removed and then test connectivity. Same for the above test case.

Reply:
We do have a test to validate both type of behaviors with and without ExternalSnat